torch/_higher_order_ops/flex_attention.py

```diff
@@ -524,8 +529,9 @@ def trace_flex_attention_backward(
         torch.zeros((), dtype=query.dtype, requires_grad=query.requires_grad)
     ] + [torch.zeros((), dtype=torch.int) for _ in range(4)]
     bw_example_vals = fw_example_vals + [torch.zeros((), dtype=query.dtype)]
-    fw_graph = make_fx(fw_graph)(*fw_example_vals, *other_buffers)
-    joint_graph = make_fx(joint_graph)(*bw_example_vals, *other_buffers)
+    with TransformGetItemToIndex():
+        fw_graph = reenter_make_fx(fw_graph)(*fw_example_vals, *other_buffers)
```

Comment:
This make_fx is invoked inside of CompositeExplicitAutograd key and it's not necessary that there's an outer make_fx session around it (for example, in eager mode).  reenter_make_fx is not applicable here because it's expecting an active parent tracer. 

Also because of the same reason, I suspect the global decompostion table is empty. That's why manually call make_fx and passing the decomp table also doesn't work. 

A fix would be 
1. check if _CURRENT_MAKE_FX_TRACER is None, if it's not None, we call reenter_make_fx.  This can happen when we're tracing the backward.
2.  if _CURRENT_MAKE_FX_TRACER is none, check whether CURRENT_DECOMPOSITION_TABLE  table is empty. if it's not empty, this means someone mannually overrides CURRENT_DECOMPOSITION_TABLE globally and we should probably respect that. 
3.  if CURRENT_DECOMPOSITION_TABLE is empty, we'll use inductors decompostion table by default, I remember it's somewhere in inductor code base.

Reply:
This is ones called from within the proxymode context, I updated it to:

```
    decomp_table = torch.fx.experimental.proxy_tensor.CURRENT_DECOMPOSITION_TABLE
    with TransformGetItemToIndex():
        fw_graph = make_fx(fw_graph, decomp_table)(*fw_example_vals, *other_buffers)
        joint_graph = make_fx(joint_graph, decomp_table)(
            *bw_example_vals, *other_buffers
        )
        
        ```
        
prior tot rebasing on the reenter pr I  had the above ^ which did resolve most of my issues:
cuurently the only thing I see, even with the reenter context is  that within the joint graph of the silu a full is not being decomposed correclty

<img width="1238" alt="Screenshot 2024-05-15 at 12 25 21 PM" src="https://github.com/pytorch/pytorch/assets/32754868/1628ddea-9925-4dab-8b34-ccd5f15da514">

Reply:
spoke offline, I am going to punt on this for now